src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/ActivityHelper.cs

```diff
@@ -35,9 +35,10 @@ internal static class ActivityHelper
         internal const string ActivityKey = "__AspnetActivity__";
 
         internal static readonly object StartedButNotSampledObj = new object();
-        private static readonly Version Version = typeof(ActivityHelper).Assembly.GetName().Version;
-        private static readonly ActivitySource AspNetSource = new ActivitySource(TelemetryHttpModule.AspNetSourceName, Version.ToString());
         private static readonly Func<HttpRequest, string, IEnumerable<string>> HttpRequestHeaderValuesGetter = (request, name) => request.Headers.GetValues(name);
+        private static readonly ActivitySource AspNetSource = new ActivitySource(
+            TelemetryHttpModule.AspNetSourceName,
+            typeof(ActivityHelper).Assembly.GetName().Version.ToString());
```

Comment:
Probably not directly related to this PR - is there a chance that this line could throw exception?

Reply:
Anything is possible 😄 I'm personally not concerned about it. Because of the `typeof(ActivityHelper).Assembly`. We are loading the assembly of the code that is running. If that assembly is unavailable for some reason, the code wouldn't be running in the first place! We would have blown up with a load exception of some kind long before this executes. But if you want me to add some error handling, happy to do it. Probably some other spots in the repo doing the same so maybe a follow-up PR?

Reply:
No need to do it in this PR.

I asked this mainly because I vaguely remember seeing exception thrown from "Assembly.GetName" in production many years ago, and I cannot remember the details. Trying to see if anyone from the community has experience dealing with such corner case.

A quick Google search "Assembly.GetName() throw exception" gives some clue that SecurityException might happen. Given here what we essentially do is `Assembly.GetExecutingAssembly().GetName()`, I guess we won't hit the problem like you (@CodeBlanch) mentioned.